### DISCUSSION TITLE
Wrong alignment of significantly shifted RS2 images

### DISCUSSION THREAD

**Discussion by AntoeLuca** [gmtsar/gmtsar#965] [Q&A] [answered]
Hello.
We are experiencing difficulties aligning two Radarsat-2 images with perpendicular baselines of approximately 100 metres, which are significantly offset along the azimuth. We have tried using XCORR as outlined in p2p_processing.csh, but have not had success. We would appreciate your assistance in resolving this issue.
![display_amp](https://github.com/user-attachments/assets/fb5a3c27-b561-4595-8662-a5ff7e5b763b)
![FootPrints](https://github.com/user-attachments/assets/678ffa6f-73c2-493a-a041-3a5f592ed479)

**Comment by Xiaohua-Eric-Xu** [comment]
Sorry for the late response. I think the problem with RS2 is that the length of orbit file is short (same length as the data) so the initial estimate of image offset from orbit info maynot be correct such xcorr could not find the right offest within the search window. 

You may 1) use precise orbit that's longer (we don't have that), or 2) cut the edge of the image (with cut_slc.c) to initially they are better aligned. Then xcorr may gets you a better aligned images.

**Reply by Xiaohua-Eric-Xu** [reply]
Most of the algorithms in GMTSAR (including focusing) rely on precise orbit (this assumption can make things run faster), so best way still is to have a precise orbit product.

**Comment by AntoeLuca** [comment]
Thank you for your reply.
Unfortunately, we do not have an accurate orbital product. We had already
attempted to use cut_slc.c before sending our request for help, but were
unsuccessful. We will try again.
In any case, xcorr functions as intended when the image offset is not so
large.

Il giorno gio 22 ago 2024 alle ore 09:30 Xiaohua(Eric) Xu <
***@***.***> ha scritto:

**Reply by Xiaohua-Eric-Xu** [reply]
A (not so great) way is to extend your orbit with extend_orbit.c and then cut your SLC to about the same area (e.g., top part of the first and bottom part of the second). Test with SAT_baseline and see whether the numbers look reasonable to you. 

This is the technique I used to process another set of data with only on-board orbit records. Hope it is helpful.

**Reply by AlexeyPechnikov** [reply]
> extend your orbit with extend_orbit.c

That is a bad way because the tool produces invalid shifted orbits as I’ve reported before. GMTSAR applies not verified 3rd party code to do the job and it does not work as expected. See the example notebook for all the details: https://github.com/AlexeyPechnikov/pygmtsar/blob/pygmtsar2/todo/orbit/read_orb_hermite.ipynb Instead, use correct Hermit interpolation (your own or realizations available in many scientific libraries) to extend orbit.

**Comment by AntoeLuca** [comment]
Thank you for your time and for your suggestion. We had already attempted
to extend the orbit and cut the images, but unfortunately without success.
It is possible that we did not do it in the most effective way. As a
temporary solution, we aligned the two images using ESA's SNAP package and
used GMTSAR for the next steps. This approach yielded satisfactory outcomes.
Thank you again for your help and for making GMTSAR available to all of us.


Il giorno sab 14 set 2024 alle ore 03:58 Xiaohua(Eric) Xu <
***@***.***> ha scritto:

**Reply by Xiaohua-Eric-Xu** [reply]
If you are able (in any case) share the two dataset, we can try to figure out a solution here.

**Comment by AntoeLuca** [comment, marked as answer]
Dear Eric,
Thank you for your message and apologies for the long delay in responding.
Unfortunately, the licence for use of the images prevents us from sharing
them. As I wrote earlier, we did the alignment with ESA-SNAP and the next
steps of analysis with GMTSAR; this "hybrid" approach seems to give good
results.
Thanks again for making gmtsar available to everyone and for your
helpfulness.

Il giorno mer 9 ott 2024 alle ore 04:50 Xiaohua(Eric) Xu <
***@***.***> ha scritto:
